bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -141,16 +133,45 @@ private String getMemberName(String topic) {
         return topicElements[3];
     }
 
-    private Object parse(String typeName, String valueToParse) throws Exception {
-        if (typeName.equals(UnDefType.class.getSimpleName())) {
+    @SuppressWarnings("unchecked")
```

Comment:
Why do you want to suppress unchecked warning?

IMHO it is much simpler to use the cast method of the given class as you already checked that is it assignable.

```java
    private <T> T parseType(final String typeName, final String valueToParse, final Class<T> desiredClass) {
        Object parsedObject = null;
        final String simpleClassName = typeName + TYPE_POSTFIX;
        parsedObject = parseSimpleClassName(simpleClassName, valueToParse);

        if (parsedObject == null || !desiredClass.isAssignableFrom(parsedObject.getClass())) {
            final String parsedObjectClassName = parsedObject != null ? parsedObject.getClass().getName()
                    : "<undefined>";
            throw new IllegalArgumentException("Error parsing simpleClasssName '" + simpleClassName + "' with value '"
                    + valueToParse + "'. Desired type was '" + desiredClass.getName() + "' but got '"
                    + parsedObjectClassName + "'.");
        } else {
            return desiredClass.cast(parsedObject);
        }
    }
```